compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -785,7 +788,11 @@ static LogicalResult setMatmulPeelingRootConfig(
   for (const auto &[index, size] : llvm::enumerate(distTileSizes)) {
     if (!size)
       continue;
-    parallelTileSizes[index] = std::min(parallelTileSizes[index], size);
+    // The backend struggles to legalize non-power-of-two scalable vectors,
+    // hence the extra rounding up.
+    parallelTileSizes[index] =
+        roundUpToPow2(std::min(parallelTileSizes[index], size),
+                      /*predicate=*/vecScalableDims[index]);
```

Comment:
Do we need this if we land https://github.com/openxla/iree/pull/16041 ?

Based on  the function name, this is used by peeling expert. We should not model any logics about masking here. So I think we can move it to below and make it:

```
// The backend struggles to legalize non-power-of-two scalable vectors,
// hence the extra rounding up.
if (hasSVE...) {
  for (auto &size : parallelTileSizes) {
    size = roundUpToPow2(size, /*predicate=*/vecScalableDims[index]);
  }
}
splitParallelAndReductionTiles(...);
```

I haven't landed my PR because there are regressions about RISC-V. And I'd like to get eyes from @dcaballe . I checked with him offline, and hopefully he can take a look soon. If you are +1 on my changes, maybe you can rebase on it and I can review this commit again.

Reply:
I've just rebased - please check the top patch and let me know whether that's what you had in mind. I do have one question though - see my comment in [select_aarch64_sve_lowering_strategy_peeling.mlir](https://github.com/openxla/iree/pull/16058/commits/75d8954c550879b5d46f1d6ad638973b0e3183b0#diff-c694d8ea45616ae2fbacd6267d761426469481ec08256c1ce53cb0998dfbf2f7).